p2p/gossip/protocol.go

```diff
@@ -245,6 +250,27 @@ func (prot *Protocol) isOldMessage(h hash) bool {
 	return oldmessage
 }
 
+func (prot *Protocol) processMessage(h hash, msg *pb.ProtocolMessage) error {
+	prot.markMessage(h)
+
+	var data service.Data
+
+	if payload := msg.GetPayload(); payload != nil {
+		data = service.DataBytes{Payload: payload}
+	} else if wrap := msg.GetMsg(); wrap != nil {
+		data = &service.DataMsgWrapper{Req: wrap.Req, MsgType: wrap.Type, ReqID: wrap.ReqID, Payload: wrap.Payload}
+	}
+
+	authKey, err := crypto.NewPublicKey(msg.Metadata.AuthPubKey)
+	if err != nil {
+		prot.Log.Error("failed to decode the auth public key when handling relay message, err %v", err)
+		return err
+	}
+
+	go prot.net.ProcessProtocolMessage(node.New(authKey, ""), msg.Metadata.NextProtocol, data)
```

Comment:
i added #442

Reply:
Not sure I understand. This PR doesn't address issue #442.